### PR TITLE
 build(docker): Configure docker setup 

### PR DESCRIPTION
**Pull request by raymurillo**
## Overview

This adds support for docker and automates the setup/configuration of the `signer` app.

Features:

- Uses localstack
- Containerizes the python development environment into the `setup` container
- Automates the configuration of localstack credentials, test certs, and KMS
- Runs the `signer` container via docker on `http://localhost:5000`

To-Dos:

-  make the `.env` path configurable in `app.py` and `setup.py`
- Add steps or a separate script to get a real cert from digicert

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.gitigonre
```
Comment (tmathern) [inline .dockerignore:1]:
```suggestion
.gitignore
```
(Typo)

File: README.md [hunk 2]
```diff
@@ -352,3 +352,21 @@ Confirm that the app signed the output image by doing one of these:
 
 - If you've installed [C2PA Tool](https://github.com/contentauth/c2patool), run `c2patool <SIGNED_FILE_NAME>.jpg`.
 - Upload the image to https://contentcredentials.org/verify. Note that Verify will display the message **This Content Credential was issued by an unknown source** because it was signed with a certificate not on the [known certificate list](https://opensource.contentauthenticity.org/docs/verify-known-cert-list).
+
+## Docker Setup
+
+### Pre-requisites
+
+- Docker Desktop
```
Comment (tmathern) [inline README.md:360]: I'd add the version you used as minimal required version

File: app.py [hunk 3]
```diff
@@ -62,9 +62,9 @@
 
     if run_mode == 'DEV':
         # For use with Localstack
-        endpoint_url = app_config['AWS_ENDPOINT']
+        endpoint_url = app_config['AWS_ENDPOINT_URL']
```
Comment (tmathern) [inline app.py:65]: Ok, so we need to change that in our dev envs too after this is merged.

Comment (tmathern) [reply app.py:65]: (Why the name change btw?)

File: local-setup.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
+echo "Creating a test user in localstack"
+awslocal --endpoint-url=http://localstack:4566 iam create-user --user-name test
+
+echo "Creating test credentials in localstack and saving them to .env.local file"
+awslocal --endpoint-url=http://localstack:4566 iam create-access-key --user-name test --output json | jq -r '.AccessKey | "AWS_ACCESS_KEY_ID=\(.AccessKeyId)\nAWS_SECRET_ACCESS_KEY=\(.SecretAccessKey)"' >> .env.local
+
+echo "Creating KMS key in localstack"
+python setup.py create-key-and-csr 'CN=John Smith,O=C2PA Python Demo'
```
Comment (tmathern) [inline local-setup.sh:10]: We could fish `'CN=John Smith,O=C2PA Python Demo'` from a setup env file var value as a next step

**Review by tmathern — APPROVED** [review]

File: app.py (hunk 3 above)
Comment (raymurillo) [reply app.py:65]: `AWS_ENDPOINT_URL` is the variable used in the [awscli](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list)

File: local-setup.sh (hunk 4 above)
Comment (gpeacock) [reply local-setup.sh:10]: If we do, it should be as a Name and an Organization field and we can construct the right string with CN= & O=

Comment (raymurillo) [reply local-setup.sh:10]: We can follow up with that.

File: README.md [hunk 5]
```diff
@@ -352,3 +352,33 @@ Confirm that the app signed the output image by doing one of these:
 
 - If you've installed [C2PA Tool](https://github.com/contentauth/c2patool), run `c2patool <SIGNED_FILE_NAME>.jpg`.
 - Upload the image to https://contentcredentials.org/verify. Note that Verify will display the message **This Content Credential was issued by an unknown source** because it was signed with a certificate not on the [known certificate list](https://opensource.contentauthenticity.org/docs/verify-known-cert-list).
+
+## Docker Setup
+
+### Pre-requisites
+
+- Docker Desktop version 4.34.3 (170107) or later
```
Comment (tmathern) [inline README.md:360]:
```suggestion
- Docker Desktop version 4.34.3 (170107) or later.
```

File: README.md [hunk 6]
```diff
@@ -352,3 +352,33 @@ Confirm that the app signed the output image by doing one of these:
...
+
+This builds and runs the containers.
+
+```shell
+make local
+```
+
+### Re-run the python client
+
+In order to re-run the python client, you can do so by running the following command:
```
Comment (tmathern) [inline README.md:372]:
```suggestion
In order to re-run the python client, run the following command:
```

**Review by tmathern — APPROVED** [review]

**Review by crandmck — APPROVED** [review]

**Pull request merged**